servers/visual/rasterizer.h

```diff
@@ -1171,6 +1171,7 @@ class RasterizerCanvas {
 				Command *n = c->next;
 				if (c == commands) {
 					memdelete(commands);
+					commands = NULL;
```

Comment:
I wonder if we should change `memdelete` to also set the pointer to `nullptr`.

Reply:
I think that setting nullptr to pointer with deleted memory should be default behaviour, because second delete on freed may cause "Double Free" errors and crashes related to it.
Freeing pointer which point on nullptr isn't a bug.
```
int* q = new int;
delete q;
delete q; //bug

int* k = new int;
delete k;
k = nullptr;
delete k; //not bug
```
I remember, that I added this once to memdelete in QtCreator, because this helped me to get info about object(I wasn't sure if object was deleted or just pointer accindentaly pointed to bad memory region)

Reply:
If @reduz confirms that this change would be good, we can find existing occurrences of resetting the pointer manually with:
```
pcregrep -M 'memdelete\(([^\)]*)\);.*\n.*\1 = (NULL|nullptr|0)' $(rg -l memdelete)
```

Reply:
`memdelete` should need to be changed to take a pointer to a pointer (`void memdelete(T **p_class)`) and you would pass such. In this case, `memdelete(&commands)`.

BTW, the nane of `p_class` is quite wrong. It should be `p_object`, but that's a separate concern.

Reply:
Here's a script which could be used to help replace all occurrences (it's pretty naive and makes mistakes which need to be reviewed manually, and also needs to be run several times as it only handles the first occurrence in each file):
```bash
#!/bin/bash

files=$(rg -l memdelete)

for file in ${files}; do
  match=$(pcregrep -M 'memdelete\(([^\)]*)\);.*\n.*\1 = (NULL|nullptr|0)' "${file}" | cut -d $'\n' -f1)
  if [ ! -z "${match}" ]; then
    sed -i "/${match}/{n;d}" "${file}"
  fi
done
```

But @reduz says this wouldn't be a good idea anyway:
```
13:13 <reduz> Akien: it does not always have write access to the pointer
13:13 <reduz> Akien: and its kind of misleading
```

Reply:
Wouldn't it be better to add at the end of template `p_class = nullptr` ?
I used this some time ago, and it seems to work fine.

https://github.com/godotengine/godot/blob/ef5891091bceef2800b4fae4cd85af219e791467/core/os/memory.h#L111-L120

like:

```
 template <class T> 
 void memdelete(T *p_class) { 
  
 	if (!predelete_handler(p_class)) 
 		return; // doesn't want to be deleted 
 	if (!__has_trivial_destructor(T)) 
 		p_class->~T(); 
  
 	Memory::free_static(p_class, false); 
        p_class = nullptr;
 } 
```

Reply:
That will just nullify the argument to `memdelete`. The original pointer won't be affected.